src/MainWindow.vala

```diff
@@ -394,6 +436,54 @@ namespace PantheonCalculator {
             entry.set_position (position);
         }
 
+        private void button_mem_clr_clicked () {
+            memory = null;
+
+            button_mem_recall.sensitive = false;
+            button_mem_clr.sensitive = false;
+        }
+
+        private void button_mem_recall_clicked () {
+            entry.set_text (memory);
+        }
+
+        private void memory_function (string functionality) {
+            var stored_mem = memory;
+            if (memory == null) {
+                stored_mem = "0";
+            }
+            var result = stored_mem + functionality + entry.get_text ();
+            var eval = new Core.Evaluation ();
+            try {
+                result = eval.evaluate (result, decimal_places);
+            } catch (Core.OUT_ERROR e) {
+                infobar_label.label = e.message;
+                infobar.no_show_all = false;
+                infobar.show_all ();
+                infobar.no_show_all = true;
+            }
+
+            memory = result;
+        }
+
+        private void button_mem_add_clicked () {
+            if (entry.get_text () != "") {
+                memory_function ("+");
+
+                button_mem_recall.sensitive = true;
+                button_mem_clr.sensitive = true;
```

Comment:
These are not needed, they will be set anyway indirectly by the `memory_function` call.